### PR TITLE
Add persistence exception handling section

### PR DESCRIPTION
**Pull request by psibi**
Let me know if this is not idiomatic. I did a related post regarding this on the mailing list: https://groups.google.com/forum/#!topic/yesodweb/MKEqVZdfam8

### CODE REVIEW

File: cookbook/Handling-Persistence-Exception.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Handling exception in persistence
+
+Say, you have a code like this which throws exception:
```
Comment (snoyberg) [inline cookbook/Handling-Persistence-Exception.md]: Perhaps "throws _an_ exception"?

File: cookbook/Handling-Persistence-Exception.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+``` haskell
+share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+User
+    age Int
+    UniqueAge age
+    deriving Show
+|]
+
+getUser :: MonadIO m => ReaderT SqlBackend m [Entity User]
```
Comment (snoyberg) [inline cookbook/Handling-Persistence-Exception.md]: Minor recommendation: `getUsers` instead of `getUser` may be a bit easier to parse for some.

**Comment by snoyberg** [comment]
There's actually a problem with this. Once an exception is thrown within a transaction, most drivers will now consider the transaction in an invalid state, and therefore further changes will fail (I think). For example, I think the code you have here will fail for PostgreSQL (though unfortunately I can't easily test that at the moment).

**Comment by gregwebs** [comment]
Yes, generally speaking a DB exception must be caught outside the transaction because the DB will not necessarily throw an exception until commit in some cases (in this case imagine another thread doing the same insertion). In this particular example one could use the PersistUnique functionality. In general sometimes (and in this case) you can break up the transaction and catch errors outside it.

**Comment by psibi** [comment]
Yes, infact the code doesn't work with the PostgreSQL backend. I have been little occupied in my day job. Will update the PR by this week.

**Comment by psibi** [comment]
I have updated the code to handle it in a database like `PostgreSQL`. Have also added a helper function `mapLiftSqlPersistMPool` which is used for executing each transaction. Probably, it can be added to `persistent` if this is the proper way to do it.

**Comment by gregwebs** [comment]
Looks better!

**Comment by snoyberg** [comment]
LGTM too.

**Comment by psibi** [comment]
Thanks for the feedback, merging it.

**Pull request merged**